ast-nodes/variable-declaration.dart

```diff
@@ -14,15 +16,32 @@ class VariableDeclaration extends Declaration {
   VariableDeclaration(name, this.type, this.value) : super(name);
 
   factory VariableDeclaration.parse(Iterable<Token> tokens) {
-    // TODO: write the actual parser body
-    return VariableDeclaration('dummy', null, Expression.parse(tokens));
+    var iter = tokens.iterator;
+    var temp_name;
+    iter.moveNext(); // Skip the `var` keyword
+    if (iter.moveNext() &&
```

Comment:
Yeah, this whole `if` is rather strange. I'd replace it with `checkNext(iter, RegExp('[A-Za-z_]\\w*\$'), "Expected a variable name")`

Reply:
`checkNext` would accept `is` in this case, which it shouldn't. Maybe it is possible to construct a RegExp that accepts all strings that consist of letters and `_` but doesn't accept  `is`, idk.